### ISSUE TITLE
Session tracking, release health not working in Managed Workflow

### ISSUE THREAD

**Issue by pweisensee** [expo/sentry-expo#181]
### Summary

I am able to track release health in a bare workflow app but not at all in my managed workflow Expo apps. 

I believe this used to work great in `sentry-expo@1.x` and maybe `sentry-expo@2.x` too. 

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android, iOS

### SDK Version (managed workflow only)

40

### Environment

```
  Expo CLI 4.2.1 environment info:
    System:
      OS: macOS 11.2.3
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 12.21.0 - ~/.nvm/versions/node/v12.21.0/bin/node
      Yarn: 1.22.10 - /usr/local/bin/yarn
      npm: 6.14.11 - ~/.nvm/versions/node/v12.21.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
      Android SDK:
        API Levels: 28, 29, 30
        Build Tools: 28.0.3, 29.0.2, 29.0.3
        System Images: android-28 | Google Play Intel x86 Atom, android-29 | Google APIs Intel x86 Atom, android-29 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: 4.0 AI-193.6911.18.40.6821437
      Xcode: 12.4/12D4e - /usr/bin/xcodebuild
    npmPackages:
      expo: ^40.0.0 => 40.0.1 
      react: 16.13.1 => 16.13.1 
      react-native: https://github.com/expo/react-native/archive/sdk-40.0.1.tar.gz => 0.63.2 
    npmGlobalPackages:
      expo-cli: 4.2.1
    Expo Workflow: managed
```

### Reproducible demo or steps to reproduce from a blank project

SDK 40 Managed Expo app, auto session tracking not working:
`sentry-expo@^3.1.3`
`@sentry/react-native@2.2.1`
```
// logging.ts
import * as Sentry from 'sentry-expo';

export const initializeSentry = () => {
    Sentry.init({
        dsn: 'https://xxxxx.ingest.sentry.io/xxxxx',
        enableAutoSessionTracking: true,  // <-- NOT WORKING
    });
};

// App.tsx
import React, { PureComponent } from 'react';
import ThemedApp from './components/ThemedApp';
import { initializeSentry } from './modules/logging';

// Initialize Sentry logging
initializeSentry();

export default class App extends PureComponent {
    render() {
        return ( <ThemedApp /> );
    }
}
```

SDK 38 Bare Expo app, auto session tracking works great:
`sentry-expo@3.0.2`
`@sentry/react-native@1.9.0`

```
// logging.ts
import * as Sentry from 'sentry-expo';

export const initializeSentry = () => {
    Sentry.init({
        dsn: 'https://xxxxx.ingest.sentry.io/xxxxx',
            enableAutoSessionTracking: true, // <-- WORKS WELL
            enableInExpoDevelopment: true, // FIX: https://github.com/expo/sentry-expo/pull/155
            environment: 'production',
            sessionTrackingIntervalMillis: 10000,
    });
};

// App.tsx
import React, { PureComponent } from 'react';
import ThemedApp from './components/ThemedApp';
import { initializeSentry } from './modules/logging';

// Initialize Sentry logging
initializeSentry();

export default class App extends PureComponent {
    render() {
        return ( <ThemedApp /> );
    }
}
```

**Comment by pweisensee** [comment]
The session auto-tracking makes a huge difference in understanding version stability and adoption rate of OTA updates. Here is the releases dashboard for both, even though they both have hundreds of active users, the managed workflow dashboard shows none.

Here is the `SDK 38` bare workflow releases dashboard:
<img width="1468" alt="Bare app SDK 38 releases" src="https://user-images.githubusercontent.com/6617776/111812142-ce79c400-88a5-11eb-96b3-0482987609ed.png">


Versus the `SDK 40` managed app releases dashboard:
<img width="1441" alt="Managed app SDK 40 releases" src="https://user-images.githubusercontent.com/6617776/111812202-e2252a80-88a5-11eb-893a-e651e42bb720.png">

**Comment by cruzach** [comment]
@pweisensee just did a little testing myself, and as I suspected, this looks to be due to the absence of the native Sentry libraries in the managed workflow (I don't think this ever worked in the managed workflow in v1 and v2, but if did then that requires more investigation)

The good news is that once EAS Build support managed workflow apps (we'll be making an announcement about that in the next month or so), those native libraries will be included in your standalone apps 😄

**Comment by pweisensee** [comment]
@cruzach okay, thanks for looking into this. Just to sum it up, this feature is not expected to work in managed workflow apps < SDK 41 because it is missing the native files and it probably never worked in Expo managed workflows in the past? 

I guess I'm excited to update to EAS once available. Thanks!

**Comment by ncoen97** [comment]
Hi, I’m using expo 41 and sentry-expo 3.1.4 in the managed workflow.
Performance monitoring and tracing are not working for me, I tryed following the sentry documentation for javascript (linked from the expo docs) and also the sentry documentation for react native and neither of those worked.
This is the code that I used while using ReactNativeTracing:

```
Sentry.init({
  dsn: 'https://...',
  enableInExpoDevelopment: true,
  debug: true,
  enableAutoSessionTracking: true,
  integrations: [
    new Sentry.Native.ReactNativeTracing({
      tracingOrigins: ['localhost', 'my-site-url.com', /^\//, /^https?:\/\//],
    }),
  ],
});
```

Is anyone else experiencing this issue?

On the other hand i have a question, is tracing suposed to work while in development (using expo go)?

Thanks in advance!
